gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionContext.java

```diff
@@ -215,6 +216,36 @@ public void removePendingSelectionSourceForOrderBy(String source) {
     pendingSelectionSourcesForOrderBy.remove(source);
   }
 
+  public void removeSelectionAttributes(String source, Set<String> attributes) {
+    if (!sourceToSelectionExpressionMap.containsKey(source)) {
+      return;
+    }
+
+    // converting to a mutable map
```

Comment:
Is the input map immutable? Why even have a immutable map when there are so many mutations happening here.

Reply:
Yes. I was planning to keep it immutable, so that any modifications done to the map is done via through the functions exposed in `ExecutionContext`, and the consumers don't accidentally/deliberately modify the map directly

Reply:
Though to keep things simple, I can change the input maps to mutable